l10n_es_genci_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2025 Ángel Rivas <angel.rivas@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    is_genci = fields.Boolean(
+        string="Apply GENCI",
+        default=False,
+        help="Indicates whether GENCI should be applied to this purchase order.",
+    )
+
+    genci_company = fields.Boolean(
+        string="Company GENCI Enabled",
+        compute="_compute_genci_company",
+    )
+
+    @api.depends("company_id.genci_enable")
+    def _compute_genci_company(self):
+        for order in self:
+            order.genci_company = bool(order.company_id.genci_enable)
+
+    @api.onchange("partner_id")
+    def _onchange_partner_genci(self):
+        for order in self:
+            if not order.partner_id:
+                order.is_genci = False
+                return
+
+            if order.partner_id.genci_subject:
+                order.is_genci = True
```

Comment:
Perform the "order.is_genci=X" operation at the very end. Do not perform multiple writes to the same field; use a temporary variable.